internal/pkg/manifest/validate.go

```diff
@@ -0,0 +1,410 @@
+	if min <= max {
+		return nil
+	}
+	return &errMinGreaterThanMax{
+		min: min,
+		max: max,
+	}
+}
+
+// Validate returns if ExecuteCommand is configured correctly.
+func (e *ExecuteCommand) Validate() error {
+	if !e.Config.IsEmpty() {
+		return e.Config.Validate()
+	}
+	return nil
+}
+
+// Validate returns if ExecuteCommandConfig is configured correctly.
+func (*ExecuteCommandConfig) Validate() error {
+	return nil
+}
+
+// Validate returns if Storage is configured correctly.
+func (s *Storage) Validate() error {
+	for k, v := range s.Volumes {
+		if err := v.Validate(); err != nil {
+			return fmt.Errorf(`validate "volumes[%s]": %w`, k, err)
+		}
+	}
+	return nil
+}
+
+// Validate returns if Volume is configured correctly.
+func (v *Volume) Validate() error {
+	if err := v.EFS.Validate(); err != nil {
+		return fmt.Errorf(`validate "efs": %w`, err)
+	}
+	return v.MountPointOpts.Validate()
+}
+
+// Validate returns if MountPointOpts is configured correctly.
+func (*MountPointOpts) Validate() error {
+	return nil
+}
+
+// Validate returns if EFSConfigOrBool is configured correctly.
+func (e *EFSConfigOrBool) Validate() error {
+	if !e.Advanced.IsEmpty() {
+		return e.Advanced.Validate()
+	}
+	return nil
+}
+
+// Validate returns if EFSVolumeConfiguration is configured correctly.
+func (e *EFSVolumeConfiguration) Validate() error {
+	if !e.EmptyBYOConfig() && !e.EmptyUIDConfig() {
+		return &errFieldMutualExclusive{
+			firstField:  "uid/gid",
+			secondField: "id/root_dir/auth",
+		}
+	}
+	if e.UID != nil && e.GID == nil {
+		return &errFieldMustBeSpecified{
+			missingField:      "gid",
+			conditionalFields: []string{"uid"},
+		}
+	}
+	if e.UID == nil && e.GID != nil {
+		return &errFieldMustBeSpecified{
+			missingField:      "uid",
+			conditionalFields: []string{"gid"},
+		}
+	}
+	if err := e.AuthConfig.Validate(); err != nil {
+		return fmt.Errorf(`validate "auth": %w`, err)
+	}
+	if aws.StringValue(e.AuthConfig.AccessPointID) != "" {
+		if (aws.StringValue(e.RootDirectory) == "" || aws.StringValue(e.RootDirectory) == "/") &&
+			aws.BoolValue(e.AuthConfig.IAM) {
+			return nil
+		}
+		return fmt.Errorf("root_dir must be either empty or / and auth.iam must be true when access_point_id is in used")
+	}
+	return nil
+}
+
+// Validate returns if AuthorizationConfig is configured correctly.
+func (*AuthorizationConfig) Validate() error {
+	return nil
+}
+
+// Validate returns if Logging is configured correctly.
+func (*Logging) Validate() error {
+	return nil
+}
+
+// Validate returns if SidecarConfig is configured correctly.
+func (s *SidecarConfig) Validate() error {
+	for ind, mp := range s.MountPoints {
+		if err := mp.Validate(); err != nil {
+			return fmt.Errorf(`validate "mount_points[%d]: %w`, ind, err)
+		}
+	}
+	return s.ImageOverride.Validate()
+}
+
+// Validate returns if NetworkConfig is configured correctly.
+func (n *NetworkConfig) Validate() error {
+	if !n.VPC.isEmpty() {
```

Comment:
I wonder if we need to check `isEmpty` before calling `Validate()`? Can the empty case be handled from within `Validate()`?